src/hotspot/cpu/ppc/macroAssembler_ppc.cpp

```diff
@@ -342,6 +342,27 @@ long MacroAssembler::get_const(address a) {
   }
   return (long) x;
 }
+// Branch-free implementation to convert !=0 to 1
+// Set register dst to 1 if dst is non-zero. Use setbcr instruction on Power10.
+void MacroAssembler::normalize_bool(Register dst, Register temp, bool is_64bit) {
+
+  if (VM_Version::has_brw()) {
+    if (is_64bit) {
+      cmpdi(CCR0, dst, 0);
+    } else {
+      cmpwi(CCR0, dst, 0);
+    }
+    setbcr(dst, CCR0, Assembler::equal);
+  } else {
+    neg(temp, dst);
+    orr(temp, dst, temp);
+    if (is_64bit) {
+      srdi(dst, temp, 63);
+    } else {
+      srwi(dst, temp, 31);
+    }
+  }
+}
```

Comment:
Please find a better place to insert this function! It doesn't fit between the 2 functions which are related.

Reply:
I am not sure how the functions have been grouped. I have grouped now based on the comment " Optimized instruction emittor" , which i think is relevant to this instruction.